tests/unittests/common/systeminfotest.cpp

```diff
@@ -103,11 +103,15 @@ TEST_F(SystemInfoTest, testIsProcessRunning) {
      QT_VERSION_CHECK(5, 3, 0))  // QProcess::processId() requires Qt>=5.3
   {
     QProcess process;
-    process.start(getTestProcessExePath().toStr());
+    QString sProgramName = getTestProcessExePath().toStr();
+    QStringList slArguments; // no args needed
+    process.start(sProgramName, slArguments);
```

Comment:
May I ask what's the intention of this change (and the same change a few lines below)? Isn't it just more verbose code with exactly the same behavior as previously on a single line?

Reply:
**primary intention:** to change it.
triggered by the warning i got in the qt ide:

![grafik](https://user-images.githubusercontent.com/5986799/104155781-2e587800-53e8-11eb-84cc-38f51a6246b8.png)

![grafik](https://user-images.githubusercontent.com/5986799/104155810-3ca69400-53e8-11eb-9819-89f5297ce4a4.png)

this was discussed in:
https://forum.qt.io/topic/116527/deprecation-of-qprocess-start-and-execute-without-argument-list

the start method with only a string as a mandatory attribute will disappear, which, in turn means this LOC will
break the build when it comes to the move to qt6, it needs to be changed some time.
(p.s. and i have overseen the third usage of process.start() that has a deprecated warning)

**secondary intention:** the way to change it.
this is very much a matter of opinion.
the shortest change would be 
    process.start(getTestProcessExePath().toStr(), QStringList());

but with this line of code my fingers instantly started to itch.
it does not really express WHAT it is doing, start a path with an undisclosed empty string list?

following some clean code advices:
introducing variables gives the chance to emphasize the intention and when needed, gives a
place to add some WHY explanations.

    QString fullProgramName = getTestProcessExePath().toStr();
    QStringList callArguments; // no args needed for this process
    process.start(sProgramName, slArguments);

the names of the local variables have been chosen according the signature of the start(...)
method. (and the hungarian notation for locals slipped-in, sorry).

there are (of course) other opinions out there about this trial to improve readability;
it is called misuse or even abuse of local variables.
that's the point where the discussions end in a dispute and should be stopped as no
party will ever agree to the argumentation of the other...

a compromise to the two versions would be the change of the way the test data is retrieved:
    process.start(getTestProgramFullName(testNr), getTestProgramArguments(testNr));

this would also give a chance to work with different test data combinations out of a set e.g. testNr = 3;

Reply:
Ah OK I see, thanks for the explanation! :+1:

Personally I'd have chosen the compact solution:

```cpp
 process.start(getTestProcessExePath().toStr(), QStringList());
```

But I'm also fine with your solution, so let's keep it :)